### PR TITLE
test_provider_ex(): Add missing call failure checks

### PR DESCRIPTION
**Pull request by t8m**
Fixes Coverity 1542440

### CODE REVIEW

File: test/provider_test.c [hunk 1]
```diff
@@ -166,11 +166,16 @@ static int test_provider_ex(OSSL_LIB_CTX **libctx, const char *name)
...
+    OSSL_PARAM_BLD *bld;
     OSSL_PARAM *params = NULL;
 
-    OSSL_PARAM_BLD_push_utf8_string(bld, "greeting", custom_buf, strlen(custom_buf));
-    params = OSSL_PARAM_BLD_to_param(bld);
+    if (!TEST_ptr(bld = OSSL_PARAM_BLD_new())
+        || !TEST_true(OSSL_PARAM_BLD_push_utf8_string(bld, "greeting", custom_buf,
+                                                      strlen(custom_buf)))
+        || !TEST_ptr(params = OSSL_PARAM_BLD_to_param(bld))) {
+        OSSL_PARAM_BLD_free(bld);
```
Comment (slontis) [inline test/provider_test.c:176]: it is probably cleaner to do this call once at the end, and init it to NULL,,

**Comment by mattcaswell** [comment]
Ping? What is happening with this PR?

File: test/provider_test.c (hunk 1 above)
Comment (t8m) [reply test/provider_test.c]: fixup pushed

**Comment by t8m** [comment]
Ping for reviews. This is needed to fix a Coverity issue.

**Review by beldmit — APPROVED** [review]
LGTM

**Review by paulidale — APPROVED** [review]

**Review by t-j-h — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**